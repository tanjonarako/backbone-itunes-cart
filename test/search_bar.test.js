import Backbone from "backbone";
import $ from "jquery";
import SearchBarView from "../src/views/search_bar";

describe('views:SearchBar', function() {
    it('call handleSubmit callback', function() {
        const spy = jest.fn();
        const searchBarView = new SearchBarView({handleSubmit: spy}).render();

        searchBarView.$el.find('input[type=text]').val('ed sheeran');
        searchBarView.$el.find('button[type=submit]').click();

        expect(spy).toBeCalledWith('ed sheeran');
        expect(searchBarView.$el.html()).toMatchSnapshot();
    });
});
