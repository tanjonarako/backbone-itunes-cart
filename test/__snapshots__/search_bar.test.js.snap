// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views:SearchBar call handleSubmit callback 1`] = `
"<form class=\\"form-inline\\">
    <div class=\\"form-group\\">
        <input class=\\"inputSearch form-control mr-1\\" type=\\"text\\" placeholder=\\"Search\\">
        <button type=\\"submit\\" class=\\"btn btn-secondary cursor-pointer\\"><i class=\\"fa fa-search\\" aria-hidden=\\"true\\"></i>Search</button>
    </div>
</form>
"
`;
